zerver/views/streams.py

```diff
@@ -374,10 +374,10 @@ def update_subscriptions_backend(
         lambda: add_subscriptions_backend(request, user_profile, streams_raw=add),
         lambda: remove_subscriptions_backend(request, user_profile, streams_raw=delete),
     ]
-    return compose_views(thunks)
+    return compose_views(request, thunks)
 
 
-def compose_views(thunks: List[Callable[[], HttpResponse]]) -> HttpResponse:
+def compose_views(request: HttpRequest, thunks: List[Callable[[], HttpResponse]]) -> HttpResponse:
```

Comment:
I think we can also avoid this API change by having this just return `json_dict`, and have the caller(s) do the `json_success` wrapping bit.